saleor/order/fetch.py

```diff
@@ -114,6 +109,7 @@ def fetch_draft_order_lines_info(
     channel = order.channel
     for line in lines:
         variant = cast(ProductVariant, line.variant)
+        product = variant.product if variant else None
```

Comment:
You have `cast` and `if variant else None`. So which one is correct here? Variant can be null or variant is always provided?

Reply:
fixed